runtime/vm/vmprops.c

```diff
@@ -494,8 +494,12 @@ addModularitySystemProperties(J9JavaVM * vm)
 		/* check if SYSPROP_JDK_MODULE_ILLEGALACCESS has been set */
 		J9VMSystemProperty *property = NULL;
 
-		if (J9SYSPROP_ERROR_NOT_FOUND != getSystemProperty(vm, SYSPROP_JDK_MODULE_ILLEGALACCESS, &property)) {
-
+		if (J9SYSPROP_ERROR_NOT_FOUND == getSystemProperty(vm, SYSPROP_JDK_MODULE_ILLEGALACCESS, &property)) {
+#if JAVA_SPEC_VERSION >= 17
+			/* beginning with Java 17, the default is --illegal-access=deny */
+			vm->runtimeFlags |= J9_RUNTIME_DENY_ILLEGAL_ACCESS;
```

Comment:
This is an unnecessary change for now.

`J9_RUNTIME_DENY_ILLEGAL_ACCESS` is not used which is being investigated by @ChengJin01 at https://github.com/eclipse/openj9/issues/11523

Reply:
That doesn't change the fact that the default is different in Java 17.
When `J9_RUNTIME_DENY_ILLEGAL_ACCESS` is used as appropriate, we will have the right behavior in all versions (including Java 17).